azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -968,4 +972,75 @@ public void cleanOlderProjectVersion(final int projectId, final int version)
       throw new ProjectManagerException("clean older project transaction failed", e);
     }
   }
+
+  @Override
+  public void uploadFlowFile(final int projectId, final int projectVersion, final int flowVersion,
```

Comment:
Will projectVersion always change when we upload a new zip?

Reply:
If the project has two ymls. If one yml changes, will only one flowVersion change?

Reply:
1. When we upload a new project zip, projectVersion will change.
2. Say the project has two flow yaml files. If one yaml file changes and user uploads a new project zip, then flowVersion will change for both yaml files. If one yaml file is changed through UI override, then we only change the flowVersion for this yaml file and won't affect the other one.